packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -468,7 +468,39 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
       } else {
         TMPL::asyncMethodCallFail(moduleName, methodName);
       }
-      throw;
+      auto exception = std::current_exception();
+      auto throwable = facebook::jni::getJavaExceptionForCppException(exception);
+      auto stackTrace = throwable->getStackTrace();
+
+      jsi::Array stackArray(runtime, stackTrace->size());
+      std::string stackTraceStr;
+      for (int i = 0; i < stackTrace->size(); ++i) {
+        auto frame = stackTrace->getElement(i);
+        stackTraceStr += frame->toString() + '\n';
+
+        jsi::Object frameObject(runtime);
+        frameObject.setProperty(runtime, "className", frame->getClassName());
+        frameObject.setProperty(runtime, "fileName", frame->getFileName());
+        frameObject.setProperty(runtime, "lineNumber", frame->getLineNumber());
+        frameObject.setProperty(runtime, "methodName", frame->getMethodName());
+        stackArray.setValueAtIndex(runtime, i, std::move(frameObject));
+      }
+
+      // TODO Better would be use getMessage() but its missing in fbjni interface
+      auto throwableString = throwable->toString();
```

Comment:
Feel free to send this as a PR to fbjni. I'm not super comfortable with the substring assumptions done below, especially as this is already on an error path.

Reply:
I've used the dynamic calls for now, same as fbjni would do, also opened the [PR there](https://github.com/facebookincubator/fbjni/pull/78), but have some issues with testing, please see the desc in the PR.

https://github.com/facebook/react-native/pull/36925/files/847043eac5132d10f05b7e58fb3e3fbaa199bede#r1169624312